### PR TITLE
wip: file upload

### PR DESCRIPTION
**Pull request by dtkav**

### CODE REVIEW

File: missioncontrol/openapi/openapi.yaml [hunk 1]
```diff
@@ -1172,26 +1172,24 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
-  /files/get_post_data_fields/:
-    get:
+  /files/presign/:
+    post:
```
Comment (Psykar) [inline missioncontrol/openapi/openapi.yaml:1176]: Yeah I was debating this one, it's certainly nicer to be able to schema it, but given it doesn't actually save anything to the DB I wasn't sure.

File: missioncontrol/v0/files.py [hunk 2]
```diff
@@ -34,23 +34,19 @@ def get(cid):
...
+def presign_upload(file_meta):
+    cid = file_meta["cid"]
     if S3File.objects.filter(cid=cid).exists():
         raise ProblemException(
-            status=409,
-            title='Conflict',
-            detail='This cid already exists in metadata',
+            status=202,
+            title='Accepted',
+            detail='File already exists in datalake, no upload is required',
```
Comment (Psykar) [inline missioncontrol/v0/files.py:50]: Fair call

**Review by Psykar — COMMENTED** [review]
Are you planning on version bump being done before merging?

**Comment by dtkav** [comment]
I'm moving this to pull against fleetspace/missioncontrol

**Pull request closed without merging**